src/main/java/pl/cleankod/exchange/entrypoint/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,16 @@
+package pl.cleankod.exchange.entrypoint;
+
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.server.exceptions.ExceptionHandler;
+import pl.cleankod.exchange.core.usecase.CurrencyConversionException;
+import pl.cleankod.exchange.entrypoint.model.ApiError;
+
+public class GlobalExceptionHandler<T extends Throwable>
+        implements ExceptionHandler<T, HttpResponse<ApiError>> {
+
+    @Override
+    public HttpResponse<ApiError> handle(HttpRequest request, T exception) {
+        return HttpResponse.badRequest(new ApiError(exception.getMessage()));
```

Comment:
Wouldn't that make every kind of exception resulting in a bad request HTTP response? How would you improve this?